mat/band.go

```diff
@@ -0,0 +1,163 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat
+
+import (
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+var (
+	bandDense *BandDense
+	_         Matrix  = bandDense
+	_         Banded  = bandDense
+	_         RawBand = bandDense
+)
+
+// BandDense represents a band matrix in dense storage format.
+type BandDense struct {
+	mat blas64.Band
+}
+
+type Banded interface {
+	Matrix
+	// Triangular returns the number of rows/columns in the matrix and its
+	// orientation.
+	Bandwidth() (kl, ku int)
+
+	// TBanded is the equivalent of the T() method in the Matrix interface but
+	// guarantees the transpose is of banded type.
+	TBanded() Banded
```

Comment:
This should be `TBand` to match `TTri` and the rest of the "abbreviation" conventions.